packages/widget/src/controllers/wallet-manager/token-balance.ts

```diff
@@ -57,12 +67,21 @@ export class TokenBalanceController implements ReactiveController {
       //resource.native is not set :shrug:
       if (resource.symbol === 'eth') {
         void this.subscribeEvmNativeBalanceUpdate();
+
         this.timeout = setInterval(
           this.subscribeEvmNativeBalanceUpdate,
           BALANCE_REFRESH_MS
         );
         return;
       }
+    } else {
+      void this.suscribeSubstrateBalanceUpdate(resource);
+      this.timeout = setInterval(
+        this.suscribeSubstrateBalanceUpdate,
+        BALANCE_REFRESH_MS,
+        resource
+      ) as unknown as ReturnType<typeof setInterval>; //dubious
```

Comment:
why do you need this cast it's already of that type

Reply:
idk why the linter doesn't like that, considering that a couple of lines above is the same declaration 🤷 . I'm going to fix it

Reply:
Looks like I needed to reload vscode